scripts/SolCx/pureshear_ALE_restart.jl

```diff
@@ -0,0 +1,215 @@
+        extrema phase ratio @ centers = $(extrema(sum(phase_ratios.center.data, dims=2)))
+    ")
+
+    # Time step
+    Δt = C * min(Δ...) / max(maximum(abs.(V.x)), maximum(abs.(V.y)))
+    @show Δt
+
+    # Create necessary tuples
+    grid_vx = (verts.x, cents_ext.y)
+    grid_vy = (cents_ext.x, verts.y)
+    Vxc     = 0.5*(V.x[1:end-1,2:end-1] .+ V.x[2:end-0,2:end-1])
+    Vyc     = 0.5*(V.y[2:end-1,1:end-1] .+ V.y[2:end-1,2:end-0])
+    Vmag    = sqrt.(Vxc.^2 .+ Vyc.^2)
+
+    for it=it0:Nt
+
+        @printf("Step %05d, #particles = %06d\n", it, Npart)
+
+        t += Δt
+
+        # advection!(particles, RungeKutta2(), values(V), (grid_vx, grid_vy), Δt)
+        # advection_LinP!(particles, RungeKutta2(), values(V), (grid_vx, grid_vy), Δt)
+        advection_MQS!(particles, RungeKutta2(), values(V), (grid_vx, grid_vy), Δt);
+        move_particles!(particles, values(verts), particle_args);
+        inject_particles_phase!(particles, phases, (), (), values(verts))
+        phase_ratios_vertex!(phase_ratios, particles, values(verts), phases);
+        phase_ratios_center!(phase_ratios, particles, values(cents), phases);  
+        Npart = sum(particles.index.data)  
+  
+        if ALE
+            xlims[1] += xlims[1]*ε̇bg*Δt 
+            xlims[2] += xlims[2]*ε̇bg*Δt
+            ylims[1] -= ylims[1]*ε̇bg*Δt 
+            ylims[2] -= ylims[2]*ε̇bg*Δt
+            L  = ( x =(xlims[2]-xlims[1]), y =(ylims[2]-ylims[1]) )  
+            Δ  = (x=L.x/Nc.x, y=L.y/Nc.y )
+            cents_ext = (
+                x      = LinRange(xlims[1]-Δ.x/2, xlims[2]+Δ.x/2, Nc.x+2),
+                y      = LinRange(ylims[1]-Δ.y/2, ylims[2]+Δ.y/2, Nc.y+2),
+            )
+            verts = (
+                x      = LinRange(xlims[1], xlims[2], Nc.x+1),
+                y      = LinRange(ylims[1], ylims[2], Nc.y+1),
+            )
+            grid_vx = (verts.x, cents_ext.y)
+            grid_vy = (cents_ext.x, verts.y)
+            Δt = C * min(Δ...) / max(maximum(abs.(V.x)), maximum(abs.(V.y)))
+            @parallel SetVelocity(V, verts, ε̇bg)
+            move_particles!(particles, values(verts), (phases,)) 
+            phase_ratios_vertex!(phase_ratios, particles, values(verts), phases)
+        end
+        
+        if mod(it, Nout) == 0 || it==1
+            @show
+            particle_density = [sum(p) for p in particles.index]
+            # visualisation
+            p    = particles.coords
+            ppx, ppy = p
+            pxv  = ppx.data[:]
+            pyv  = ppy.data[:]
+            clr  = phases.data[:]
+            idxv = particles.index.data[:]
+            f = Figure()
+            ax = Axis(f[1, 1], title="Particles", aspect=L.x/L.y)
+            scatter!(ax, Array(pxv[idxv]), Array(pyv[idxv]), color=Array(clr[idxv]), colormap=:roma, markersize=2)
+            xlims!(ax, verts.x[1], verts.x[end])
+            ylims!(ax, verts.y[1], verts.y[end])
+            display(f)
+        end       
+
+        # Save breakpoint
```

Comment:
```suggestion
        # Save checkpoint
```

Reply:
? I have always called that a breakpoint

Reply:
It's usually called [checkpoint](https://en.wikipedia.org/wiki/Application_checkpointing)